semgrep/semgrep/commands/parse.py

```diff
@@ -0,0 +1,84 @@
+import os
+from typing import Sequence
+from typing import Tuple
+
+import click
+
+import semgrep.semgrep_main
+from semgrep.constants import OutputFormat
+from semgrep.constants import RuleSeverity
+from semgrep.error import SemgrepCoreError
+from semgrep.error import SemgrepError
+from semgrep.formatter.text import TextFormatter
+from semgrep.output import OutputHandler
+from semgrep.output import OutputSettings
+from semgrep.rule_match import RuleMatch
+from semgrep.verbose_logging import getLogger
+
+logger = getLogger(__name__)
+
+
+@click.command()
+@click.option("--lang", "-l", help="Parse pattern and all files in specified language.")
+@click.argument("target", nargs=-1, type=click.Path(allow_dash=True))
+def parse(lang: str, target: Tuple[str, ...]) -> None:
+    """
+    Tries to parse all files. Shows which files have problems parsing in language
+    """
+    semgrep.util.set_flags(verbose=False, debug=False, quiet=False, force_color=False)
+    target_sequence: Sequence[str] = list(target) if target else [os.curdir]
+    pattern = "$X = 12345"
```

Comment:
maybe a comment about this pattern :)
Also maybe a comment next to optimizations="none" which is quite important.